stdlib/asm/crypto/hashes/blake3.masm

```diff
@@ -754,3 +793,51 @@ export.hash.4
 
     exec.finalize
 end
+
+#! Blake3 2-to-1 hash function, which takes 32 -bytes input and produces 32 -bytes output digest
+#!
+#! Expected stack state:
+#!
+#! [msg0, msg1, msg2, msg3, msg4, msg5, msg6, msg7, ...]
+#!
+#! msg`i` -> 32 -bit message word | i ∈ [0, 8)
+#!
+#! Final stack state:
+#!
+#! [dig0, dig1, dig2, dig3, dig4, dig5, dig6, dig7, ...]
+#!
+#! dig`i` -> 32 -bit digest word | i ∈ [0, 8)
+export.hash32.4
```

Comment:
A couple of comments here:

1. This is no longer a `2-to-1` hash but rather a `1-to-1` hash.
2. I wonder if a better name would be `hash_1x1` and then `hash64` can also be renamed into `hash_2x1`.

Also a comment for the future: we usually use a slightly different convention for arranging elements on the stack. Specifically, we usually put the "least significant" value deep into the stack. Under this convention, the inputs would look like so:
```
[msg7, msg6, msg5, msg4, msg3, msg2, msg1, msg0, ...]
```
And the outputs would be:
```
[dig7, dig6, dig5, dig4, dig3, dig2, dig1, dig0, ...]
```
This would also mean that when providing stack inputs, we don't need to reverse them, but we would need to reverse the outputs when reading back from the stack.

Let's create an issue to update this in the future.

Reply:
Speaking of naming, I've changed their names to 

hash32 -> hash_1to1
hash64 -> hash_2to1

I don't have any objection about `hash_{1,2}x1` either.

Reply:
Speaking of data ordering on stack, I've taken a note down and I'll put it in an issue. I'll link that here in sometime.

Reply:
Created https://github.com/maticnetwork/miden/issues/571 so that's we don't forget about it.

Reply:
> Speaking of naming, I've changed their names to
> 
> hash32 -> hash_1to1 hash64 -> hash_2to1

I like these!